keps/sig-network/3015-prefer-same-node/README.md

```diff
@@ -0,0 +1,525 @@
+
+(User workloads that use the feature may expose SLI information that
+the user can examine to determine how well the feature is working for
+their workload.)
+
+###### Are there any missing metrics that would be useful to have to improve observability of this feature?
+
+Not really; we don't know how fast the user's services are supposed to
+be, so we can't really tell if we are improving them as much as they
+hoped or not.
+
+### Dependencies
+
+###### Does this feature depend on any specific services running in the cluster?
+
+Any service proxy that implemented `PreferClose` in the "standard" way
+will automatically also handle `PreferSameZone`, because the proxy
+does not actually look at `TrafficDistribution` itself; it only looks
+at the EndpointSlice hints, which are set by kube-controller-manager.
+
+The `PreferSameNode` semantics will require a service proxy that has
+been updated to know about the `ForNodes` hint. We will update
+`kube-proxy` ourselves, but network plugins / kubernetes distributions
+that ship their own alternative service proxies will also need to be
+updated to support the new value before their users can make use of
+it. (Until then, `TrafficDistribution: PreferSameNode` would end up
+falling back to the semantics of `TrafficDistribution:
+PreferSameZone`.)
+
+### Scalability
+
+###### Will enabling / using this feature result in any new API calls?
+
+No
+
+###### Will enabling / using this feature result in introducing new API types?
+
+No
+
+###### Will enabling / using this feature result in any new calls to the cloud provider?
+
+No
+
+###### Will enabling / using this feature result in increasing size or count of the existing API objects?
+
+No (other than that it means people may set `TrafficDistribution` on
+Services where they were not previously setting it).
+
+###### Will enabling / using this feature result in increasing time taken by any operations covered by existing SLIs/SLOs?
+
+No
+
+###### Will enabling / using this feature result in non-negligible increase of resource usage (CPU, RAM, disk, IO, ...) in any components?
+
+No
+
+### Troubleshooting
+
+###### How does this feature react if the API server and/or etcd is unavailable?
+
+No change from existing service/proxy behavior.
+
+###### What are other known failure modes?
+
+None known
+
+###### What steps should be taken if SLOs are not being met to determine the problem?
+
+N/A
+
+## Implementation History
+
+- Initial proposal as `InternalTrafficPolicy: PreferLocal`: 2021-10-21
+- Initial proposal as "Node-level topology": 2022-01-15
+- Initial proposal as `TrafficDistribution: PreferSameNode`: 2025-02-06
+- Added `TrafficDistribution: PreferSameZone`: 2025-02-08
+
+## Drawbacks
+
+## Alternatives
+
+"Node-level topology" was one of many things that was possible with
+the old [`TopologyKeys`] API, but that API was deprecated because of
+other problems.
+
+This is the third attempt at "node-level topology" specifically.
```

Comment:
```suggestion
_Node-level topology_ was one of many things that was possible with
the old [`TopologyKeys`] mechanism, but that enhancement to the Service API was deprecated
(and removed) because of other problems.

This is the third attempt at _node-level topology_ specifically.
```
?

Reply:
I'm not really defining this (or "topologically available") as a "concept" though. Like, if there was a glossary, neither of them would be included there. I'll just remove the quotes.